app/models/course/material/folder.rb

```diff
@@ -93,6 +93,13 @@ def next_valid_name
     parent.next_uniq_child_name(self)
   end
 
+  # Take Course#advance_start_at_duration into account when calculating folder's start datetime.
+  #
+  # @return [DateTime] The shifted start_at datetime.
+  def effective_start_at
+    start_at - (course&.advance_start_at_duration || 0)
```

Comment:
Suggest we move the default value of 0 into the course model.

Reply:
I tried this, but there are side effects. On the course settings page, SDL is now always treated as set.

The course settings page depends on `course.advance_start_at_duration` to return nil if the setting is not set. If that view is changed to check for 0 instead, an instructor could lock himself out of the setting by setting SDL to 0.

Reply:
Then to be fair this should be contained within the course model, and not have the implementation detail spill out to other models? 

Are there any ways of fixing this?

Reply:
Fixed already. There's already an existing check that prevents SDL from being set to 0, so it's safe to use that as a proxy for set/unset.